### DISCUSSION TITLE
Would the sveltekit form action still work?

### DISCUSSION THREAD

**Discussion by leovoon** [icflorescu/trpc-sveltekit#49] [Q&A]
I'm getting ``Error: Invalid export 'actions' (valid exports are load, prerender, csr, ssr, trailingSlash, or anything with a '_' prefix)`` when export actions from `+page.ts`.

While in `+page.server.ts`, there is no error. But I could not get the url.searchParams

```js
// +page.server.ts
export const actions: Actions = {
  default: async ({ url }) => {
    console.log(url.searchParams.get('email')) // null
  },
}
```

```html
// +page.svelte
<form method="POST">
  <input type="text" name="email" />
  <button type="submit">submit</button>
</form>
```

**Comment by icflorescu** [comment]
This has nothing to do with `trpc-sveltekit`.
Please consult the SvelteKit docs page: https://kit.svelte.dev/docs/form-actions
Just a couple of hints, though:
If you're using `POST`, `email` can't be a search param; you'll probably have to extract it from request forma data.
Also, form actions have nothing to do with `+page.ts`. They're handled in `+page.server.ts`.

**Reply by leovoon** [reply]
I misunderstood the concept. My apologies. Thanks for your answer!

**Reply by icflorescu** [reply]
No worries.
Please note that you definitely COULD use a tRPC procedure to handle form data.
Have a look at the Bookstall example in the repo.
